scripts/lib/install

```diff
@@ -325,102 +323,70 @@ fi
-        if ! dpkg-query --showformat '${Status}\n' -W rabbitmq-server 2>/dev/null | grep -vq ' not-installed$'; then
-            cat <<EOF
+[postgresql]
+version = $POSTGRES_VERSION
 
 [rabbitmq]
 nodename = zulip@localhost
 EOF
-        fi
-
-        if [ -n "$USE_CERTBOT" ]; then
-            cat <<EOF
-
-[certbot]
-auto_renew = yes
-EOF
-        fi
 
-        case ",$PUPPET_CLASSES," in
-            *,zulip::voyager,* | *,zulip::postgres_appdb_tuned,*)
-                if [ "$package_system" = apt ]; then
-                    cat <<EOF
+    if [ -n "$USE_CERTBOT" ]; then
+        crudini --set /etc/zulip/zulip.conf certbot auto_renew yes
+    fi
 
-[postgresql]
-version = $POSTGRES_VERSION
-EOF
-                fi
-                ;;
-        esac
-    ) > /etc/zulip/zulip.conf
-fi
+    if [ -n "$POSTGRES_MISSING_DICTIONARIES" ]; then
+        crudini --set /etc/zulip/zulip.conf postgresql missing_dictionaries true
+    fi
 
-case ",$PUPPET_CLASSES," in
-    *,zulip::voyager,* | *,zulip::dockervoyager,* | *,zulip::app_frontend,*)
-        if [ -z "$NO_OVERWRITE_SETTINGS" ] || ! [ -e "/etc/zulip/settings.py" ]; then
-            cp -a "$ZULIP_PATH"/zproject/prod_settings_template.py /etc/zulip/settings.py
-            if [ -n "$EXTERNAL_HOST" ]; then
-                sed -i "s/^EXTERNAL_HOST =.*/EXTERNAL_HOST = '$EXTERNAL_HOST'/" /etc/zulip/settings.py
-            fi
-            if [ -n "$ZULIP_ADMINISTRATOR" ]; then
-                sed -i "s/^ZULIP_ADMINISTRATOR =.*/ZULIP_ADMINISTRATOR = '$ZULIP_ADMINISTRATOR'/" /etc/zulip/settings.py
-            fi
-        fi
-        ln -nsf /etc/zulip/settings.py "$ZULIP_PATH"/zproject/prod_settings.py
-        "$ZULIP_PATH"/scripts/setup/generate_secrets.py --production
-        ;;
-esac
+    "$ZULIP_PATH"/scripts/zulip-puppet-apply --force --noop \
+                 --write-catalog-summary \
+                 --classfile=/var/lib/puppet/classes.txt \
+                 >/dev/null
 
-"$ZULIP_PATH"/scripts/zulip-puppet-apply -f
+    if ! has_class "zulip::postgres_common" || [ "$package_system" != apt ]; then
+        crudini --del /etc/zulip/zulip.conf postgresql
+    fi
 
-if [ "$package_system" = apt ]; then
-    SUPERVISOR_CONF_DIR="/etc/supervisor/conf.d"
-elif [ "$package_system" = yum ]; then
-    SUPERVISOR_CONF_DIR="/etc/supervisord.d/conf.d"
+    if ! has_class "zulip::rabbit"; then
+        crudini --del /etc/zulip/zulip.conf rabbitmq
+    fi
 fi
 
-# Detect which features were selected for the below
-set +e
-[ -e "/etc/init.d/nginx" ]; has_nginx=$?
-[ -e "$SUPERVISOR_CONF_DIR/zulip.conf" ]; has_appserver=$?
-[ -e "/etc/cron.d/rabbitmq-numconsumers" ]; has_rabbit=$?
-[ -e "/etc/init.d/postgresql" ]; has_postgres=$?
-set -e
-
-# Docker service setup is done in the docker config, not here
-if [ "$DEPLOYMENT_TYPE" = "dockervoyager" ]; then
-    has_nginx=1
-    has_appserver=0
-    has_rabbit=1
-    has_postgres=1
+if has_class "zulip::app_frontend"; then
```

Comment:
I think this should be `zulip::app_frontend_base`.  The logic is basically always skipped for the `zulip_ops` subclasses of it because we use `NO_OVERWRITE_SETTINGS` there, so it doesn't matter for existing subclasses, but I think `app_frontend_base` is the check for whether something is an application server, which is what we're replicating here.